src/http_upload/mod_http_upload.erl

```diff
@@ -110,6 +104,27 @@ iq_handler(_From, _To = #jid{lserver = SubHost}, Acc, IQ = #iq{type = get, sub_e
     end,
     {Acc, Res}.
 
+-spec disco_iq_handler(From :: jid:jid(), To :: jid:jid(), Acc :: mongoose_acc:t(),
+                       IQ :: jlib:iq()) ->
+          {mongoose_acc:t(), jlib:iq()}.
+disco_iq_handler(_From, _To, Acc, #iq{type = set, sub_el = SubEl} = IQ) ->
+    {Acc, IQ#iq{type = error, sub_el = [SubEl, mongoose_xmpp_errors:not_allowed()]}};
+disco_iq_handler(_From, To, Acc, #iq{type = get, lang = Lang, sub_el = SubEl} = IQ) ->
+    LServer = To#jid.lserver,
```

Comment:
could be matched in the function clause.

Reply:
I'd rather limit the clause head to fit in one line less than 100 characters.